test/unit/metadata.js

```diff
@@ -772,4 +774,67 @@ describe('metadata', function () {
       should(upToDate(doc).errors).be.undefined()
     })
   })
+
+  describe('generated a doc with conflict', () => {
+    it('should get the correct path', () => {
+      const doc = {
+        path: 'docname'
+      }
+      const newDoc = createConflictingDoc(doc)
+      const pathRegExp = withConflictRegExp(doc.path)
+      should(newDoc.path).match(pathRegExp)
+    })
+    it('should get the correct _id', () => {
+      const doc = {
+        path: 'docname'
+      }
+      const newDoc = createConflictingDoc(doc)
+      const pathRegExp = withConflictRegExp(doc.path)
+      should(newDoc._id).match(pathRegExp)
+    })
+    it('should keep the correct extension', () => {
+      const ext = '.pdf'
+      const doc = {
+        path: `docname${ext}`
+      }
+      const newDoc = createConflictingDoc(doc)
+      should(path.extname(newDoc.path)).equal(ext)
+    })
+    it('should keep the correct extension when extension is complex', () => {
+      const ext = '.tar.gz'
+      const doc = {
+        path: `docname${ext}`
+      }
+      const newDoc = createConflictingDoc(doc)
+      // FIXME: must be docname-conflict-:ISODATE:.tar.gz instead of docname.tar-conflict-:ISODATE:.gz
+      should(path.extname(newDoc.path)).equal('.gz')
+    })
+    it('should not have more than 180 characters', () => {
+      const doc = {
+        path: 'Lorem ipsum dolor sit amet consectetur adipiscing elit Nam a velit at dolor euismod tincidunt sit amet id ante Cras vehicula lectus purus In lobortis risus lectus vitae rhoncus quam porta nullam'
+      }
+      const newDoc = createConflictingDoc(doc)
+      const newDocBasename = withConflictRegExp('(.*)').exec(path.basename(newDoc.path))[1]
```

Comment:
Looks a bit like lines 482 to 484 in the implementation?

Reply:
I will check it.

Reply:
Well, it doesn't matter so much. Please forget about my comment :)